From 66f84ee8cecd5501bf954a633a778a07856f81c5 Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Wed, 11 Jul 2018 16:41:42 +0300
Subject: [PATCH 4157/5242] MLK-18850: drm/bridge: nwl-dsi: Fix bridge
 handling

commit  9b46da3f737cee25c36ccdbcc507a86c4a543ed8 from
https://source.codeaurora.org/external/imx/linux-imx.git

When the drm_bridge object, representing the next bridge in chain,
connected to nwl-dsi brigde was attached to encoder, the
"previous_bridge" was wrongly passed, this way the next bridge was not
actually added to the encoder bridge chain.

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/bridge/nwl-dsi.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/bridge/nwl-dsi.c b/drivers/gpu/drm/bridge/nwl-dsi.c
index c6a8caa..9ac8447 100644
--- a/drivers/gpu/drm/bridge/nwl-dsi.c
+++ b/drivers/gpu/drm/bridge/nwl-dsi.c
@@ -960,7 +960,7 @@ static int nwl_dsi_bridge_attach(struct drm_bridge *bridge)
 		}
 
 		dsi->next_bridge = of_drm_find_bridge(remote_node);
-		ret = drm_bridge_attach(encoder, dsi->next_bridge, encoder->bridge);
+		ret = drm_bridge_attach(encoder, dsi->next_bridge, bridge);
 		if (ret)
 			dsi->next_bridge = NULL;
 		of_node_put(remote_node);
-- 
1.7.9.5

